### PR TITLE
ZENKO-2848 skip end2end tests on documentation content

### PR DESCRIPTION
**Pull request by tcarmet**
In this PR we're reorganizing the pipeline so that the stages
that triggers the end2end tests are completely skipped at CI level
when the content of a branch is documentation related only.

### CODE REVIEW

File: eve/main.yml [hunk 1]
```diff
@@ -189,12 +189,25 @@ stages:
     worker:
       type: local
     steps:
+    # Checking weither the content of the branch is only documentation related.
```
Comment (alexanderchan-scality) [inline eve/main.yml]: weither => whether

File: eve/main.yml [hunk 2]
```diff
@@ -189,12 +189,25 @@ stages:
...
+    # This will help in deciding wheither the end2end tests
```
Comment (alexanderchan-scality) [inline eve/main.yml]: Whether => whether

File: eve/scripts/launch-end2end.sh [hunk 3]
```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+# Helper script that check if the content and name of the branch is documentation related only.
+# This help us deciding wheither we should or not launch the end2end tests.
```
Comment (alexanderchan-scality) [inline eve/scripts/launch-end2end.sh]: Wheither => whether

**Review by alexanderchan-scality — COMMENTED** [review]
Just a few minor changes; lgtm otherwise

File: eve/main.yml (hunk 1 above)
Comment (tcarmet) [reply eve/main.yml]: hahaha found a way to make different mistakes on the same word. Thanks !

File: eve/main.yml (hunk 2 above)
Comment (tcarmet) [reply eve/main.yml]: done

File: eve/scripts/launch-end2end.sh (hunk 3 above)
Comment (tcarmet) [reply eve/scripts/launch-end2end.sh]: done

**Comment by tcarmet** [comment]
@alexanderchan-scality I was thinking about reverting this change for 2.0, but what's your thoughts on this, do you think I should keep it and adapt it for development/2.x branches or remove it?

File: eve/scripts/launch-end2end.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#!/bin/sh
+
+# Helper script that check if the content and name of the branch is documentation related only.
+# This help us deciding whether we should or not launch the end2end tests.
+# Returns true when it's not documentation related and false when it is. 
+
+
+if git rev-parse --abbrev-ref HEAD | grep -q 'documentation/*' ; then
+    diff="$(git diff --name-only HEAD~1 | grep -v docs/)"
```
Comment (miniscruff) [inline eve/scripts/launch-end2end.sh]: might be valuable to move the `docs/` grep value to a variable as I feel there are more instances that we wouldn't need to run e2e ( or any tests? ). Such as the .github folder, res folder, readme.

File: eve/scripts/launch-end2end.sh [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    if [ ! -z "$diff" ]; then
+        echo "true"
```
Comment (miniscruff) [inline eve/scripts/launch-end2end.sh]: Not sure the quotes are required but I do not know how eve handles this.

File: eve/scripts/launch-end2end.sh (hunk 4 above)
Comment (tcarmet) [reply eve/scripts/launch-end2end.sh]: Good idea, gonna add that.

File: eve/scripts/launch-end2end.sh (hunk 5 above)
Comment (tcarmet) [reply eve/scripts/launch-end2end.sh]: They're not, I can remove them

Comment (tcarmet) [reply eve/scripts/launch-end2end.sh]: done

**Review by miniscruff — APPROVED** [review]

**Comment by alexanderchan-scality** [comment]
> @alexanderchan-scality I was thinking about reverting this change for 2.0, but what's your thoughts on this, do you think I should keep it and adapt it for development/2.x branches or remove it?

@tcarmet I believe the changes made here can be kept. The benefit from skipping e2e tests for documentation changes make sense for 2.x as well. Correct me if I am wrong, I don't seem much that needs to be done to adapt it for 2.x.

**Comment by tcarmet** [comment]
> > @alexanderchan-scality I was thinking about reverting this change for 2.0, but what's your thoughts on this, do you think I should keep it and adapt it for development/2.x branches or remove it?
> 
> @tcarmet I believe the changes made here can be kept. The benefit from skipping e2e tests for documentation changes make sense for 2.x as well. Correct me if I am wrong, I don't seem much that needs to be done to adapt it for 2.x.

Alright, thanks for the input. I'll add in 2.x as well. And yes, it's easy enough to adapt for 2.x

**Comment by tcarmet** [comment]
made the appropriate changes for w/2.0 @alexanderchan-scality 
[have a look](https://github.com/scality/zenko/compare/development/2.0...w/2.0/bugfix/ZENKO-2848-disable-e2e-tests-documentation-pr) to review this pr properly

**Review by naren-rajendran — APPROVED** [review]

**Review by anurag4DSB — APPROVED** [review]

**Comment by anurag4DSB** [comment]
CC @wabernatScality for visibility

**Pull request merged**